src/ShiftedCompositeNormL1.jl

```diff
@@ -0,0 +1,85 @@
+export ShiftedCompositeNormL1
+
+mutable struct ShiftedCompositeNormL1{
+  R <: Real,
+  V0 <: Function,
+  V1 <: Function,
+  V2 <: AbstractMatrix{R},
+  V3 <: AbstractVector{R},
+  V4 <: AbstractVector{R}
+} <: ShiftedCompositeProximableFunction
+  h::NormL1{R}
+  c!::V0
+  J!::V1
+  A::V2
+  b::V3
+  sol::V4
+  function ShiftedCompositeNormL1(
+    h::NormL1{R},
+    c!::Function,
+    J!::Function,
+    A::AbstractMatrix{R},
+    b::AbstractVector{R},
+  ) where {R <: Real}
+    sol = similar(b,size(A,2))
+    if length(b) != size(A,1)
+      error("ShiftedCompositeNormL1 : Wrong input dimensions, constraints should have same length as rows of the Jacobian")
+    end
+    new{R,typeof(c!),typeof(J!),typeof(A),typeof(b), typeof(sol)}(h,c!,J!,A,b, sol)
+  end
+end
+
+
+shifted(h::NormL1{R}, c!::Function,J!::Function,A::AbstractMatrix{R},b::AbstractVector{R}, xk :: AbstractVector{R}) where {R <: Real} = begin
+  c!(b,xk)
+  J!(A,xk)
+  ShiftedCompositeNormL1(h,c!,J!,A,b)
+end
+
+shifted(
```

Comment:
What is this expected to return? $t \to \Vert c(x) + J(x) (s + t) \Vert_1$? If so, why do you need to evaluate $c(x)$ and $J(x)$ again?

Reply:
This centers the shift to $t$, `ShiftedCompositeNormL1` represents $s \xrightarrow[]{} ||c(x) + J(x)s||_1$. When we apply `shifted`, we get  $s \xrightarrow[]{} ||c(t) + J(t)s||_1$. This is similar to what is done on usual `ShiftedProximableFunction`.